app/controllers/optIn/CheckYourAnswersController.scala

```diff
@@ -62,10 +63,15 @@ class CheckYourAnswersController @Inject()(val view: CheckYourAnswersView,
       withRecover(isAgent) {
 
         val result = for {
-          taxYear <- OptionT(optInService.fetchSavedChosenTaxYear())
+          intentTaxYear <- OptionT(optInService.fetchSavedChosenTaxYear())
           cancelURL = ReportingFrequencyPageController.show(isAgent).url
-          intentIsNextYear = taxYear.isNextTaxYear(dateService)
-        } yield Ok(view(MultiYearCheckYourAnswersViewModel(taxYear, isAgent, cancelURL, intentIsNextYear)))
+          intentIsNextYear = intentTaxYear.isNextTaxYear(dateService)
```

Comment:
Ian: "Could we move this logic out of the controller and instead build a view model in the Service? This will make testing this logic much simpler."

Reply:
@ian-hmrc Done

Reply:
Cool